src/main/java/core/basesyntax/strategy/handler/BalanceHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Balance cannot be negative");
+        }
+        fruit.setQuantity(quantity);
+        Storage.addOrUpdateFruit(fruit.getName(), fruit);
```

Comment:
To follow the encapsulation principle, the `Storage` class should not expose its implementation. Instead, it should provide methods to interact with the storage without revealing the details of how the data is stored.

Reply:
The logic was refactored